### PR TITLE
[Fixes #7325] Add EIP681 deeplink support

### LINKED ISSUE #7325: EIP681 URLs direct to Status
### User Story

As a user, I want to have EIP681 style URI links redirected to Status.

### Description

*Type*: Feature

*Summary*: Formerly EIP67, now [681](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-681.md).

Currently blocked by PR #7263  

#### Expected behavior
Allows Ethereum transactions to be represented by a URL or QR code, which then open in Status.

### PR DESCRIPTION
**Pull request by jeluard**
fixes #7325

### Summary

Added EIP681 deep link support.

_NOTE to testers_ [EIP681](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-681.md) clicked in mobile stock browser should open status / send screen.

status: ready

### CODE REVIEW

**Review by goranjovic — APPROVED** [review]

**Review by flexsurfer — APPROVED** [review]

**Review by rasom — APPROVED** [review]

**Comment by churik** [comment]
@jeluard please, rebase to current `develop`.
Thanks!

**Comment by jeluard** [comment]
@churik Done!

File: src/status_im/ui/screens/wallet/choose_recipient/events.cljs
Comment (yenda) [inline src/status_im/ui/screens/wallet/choose_recipient/events.cljs]: what is this change  about?

**Review by yenda — APPROVED** [review]

File: src/status_im/ui/screens/wallet/choose_recipient/events.cljs
Comment (jeluard) [reply src/status_im/ui/screens/wallet/choose_recipient/events.cljs]: Wallet flow hardcode some `navigate-back` that are not wanted here.

Comment (yenda) [reply src/status_im/ui/screens/wallet/choose_recipient/events.cljs]: can you comment it for the glorious day we'll refactor wallet flow?

Comment (jeluard) [reply src/status_im/ui/screens/wallet/choose_recipient/events.cljs]: :+1:

**Comment by churik** [comment]
@jeluard 
Please, add a tests that you performed on your devices, as we agreed before.
In this particular case I would like to know examples of the links which you were trying and browsers \ ways how you navigate to the app.

Please, elaborate:
1) I can't find a way to use it as a link, i.e. if I try to open `ethereum:0x744d70fdbe2ba4cf95131626614a1763df805b9e/transfer?address=0x8e23ee67d1332ad560396262c48ffbb01f93d052&uint256=1` in Safari (IOS) or inside Status, I've got 'Unable to load page'.
Did I do smth wrong?

I tested the following scenarios using "Scan QR" on home screen (generating them using https://www.qr-code-generator.com/ > Text) or via Wallet > Send > Scan QR (result is that same)

2)  Links without `function_name` are not resolved, i.e. `ethereum:0xfb6916095ca1df60bb79Ce92ce3ea74c37c5d359?value=2.014e18` so when scanning URL with this link, got an error 
Monosnap 2019-05-01 11-19-13

3) If using ERC #20 tokens as parameters, value is not resolved, i.e. after scanning `ethereum:0x744d70fdbe2ba4cf95131626614a1763df805b9e/transfer?address=0x8e23ee67d1332ad560396262c48ffbb01f93d052&uint256=1` should sent 1 SNT as far as I understand.
In fact after scanning QR with link I've got:
photo_2019-05-01 11 27 14

So the asset was changed to tokens, but the value is not displayed.

4) Parameters `gasLimit` and `gasPrice` are not resolved (i.e. in `ethereum:0x744d70fdbe2ba4cf95131626614a1763df805b9e/transfer?address=0x8e23ee67d1332ad560396262c48ffbb01f93d052&value=1&gasLimit=100&gasPrice=1`
However `gas` is resolved, as it mentioned in [spec](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-681.md)

**Comment by jeluard** [comment]
Sure, will describe how to test this.

Are issue 2,3,4 regression from develop? This PR doesn't attempt to change existing EIP support, just make it accessible to deep links.

**Comment by churik** [comment]
@jeluard  do you mean https://github.com/status-im/status-react/pull/2430 PR?
I believe there were tested only QRs, given in PR.
So, in this case, we have several issues according to current spec of [EIP 681](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-681.md).
Should they be investigated \ reported separately?

**Comment by jeluard** [comment]
Ahah looks like you enjoy history!

**Comment by jeluard** [comment]
We will put this PR on hold for now as it requires more work.

**Comment by rachelhamlin** [comment]
@jeluard @churik  is there something worth merging from this PR? Should we merge & open separate issues?

**Comment by jeluard** [comment]
WDYT @churik ? If there are no regression from develop we could merge it and revisit later?

**Comment by churik** [comment]
@jeluard @rachelhamlin I don't see what exactly is implemented in this PR and didn't see the way to test it, so don't see the reason to merge this PR unless you elaborate on test cases and value for the end user.
Thanks.

**Comment by jeluard** [comment]
@churik What changed is the deep link support. I will provide some instruction on how to test that

**Comment by jeluard** [comment]
@churik PR updated. It can now be tested through a new tab in simpledapp (changes yet to be published)

**Comment by jeluard** [comment]
@churik You can test using http://condescending-jones-85ad1f.netlify.com
Click on the link on a regular mobile browser, it should open status wallet screen.

**Comment by churik** [comment]
@jeluard please, rebase PR to current `develop` - some of failed tests should be fixed.

Also some notes: 
1) When I open deep link from http://condescending-jones-85ad1f.netlify.com/ I can proceed to `Send wallet` even if I didn't setup wallet yet.
So steps are:
- create account
- without setting up account put app to background
- open http://condescending-jones-85ad1f.netlify.com and tap on the link

2) additional parameters for full support EIP681 will be added separately? should I create issue for this? 
So, for now, I'm testing only `address` (according to https://github.com/status-im/status-react/pull/8004#issuecomment-489562602)

**Comment by jeluard** [comment]
@churik Done

**Comment by churik** [comment]
@jeluard any comments on https://github.com/status-im/status-react/pull/8004#issuecomment-489606688?

**Comment by jeluard** [comment]
@churik Sorry missed that comment!

1) is definitely an issue I need to fix
2) yes please open a separate issue for that

**Comment by jeluard** [comment]
@churik 1) should be fixed, with the caveat that onboarding will loose the wallet details. User would have to  click a second time on the deep link.

**Comment by churik** [comment]
Created #8150 
Tested on IOS and Android as it mentioned in https://github.com/status-im/status-react/pull/8004#issuecomment-489562602
If user should create an account or pass through set up process - data from deep link is lost, it discussed and decided that it is expected at the moment.

**Comment by churik** [comment]
`test_send_two_transactions_one_after_another_in_dapp` retested manually - false failure

**Pull request merged**
